tests/testsuite/standard_lib.rs

```diff
@@ -366,16 +366,113 @@ fn target_proc_macro() {
             "src/lib.rs",
             r#"
                 extern crate proc_macro;
-                pub fn f() {
+                fn f() {
                     let _ts = proc_macro::TokenStream::new();
                 }
             "#,
         )
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "pm"
+                version = "0.1.0"
+                [lib]
+                proc-macro = true
+            "#,
+        )
         .build();
 
     p.cargo("build -v").build_std(&setup).target_host().run();
 }
 
+// We already have `basic` which uses `proc_macro::custom_api()`. This case attempts to use
+// `TokenStream` which would error because we are using the sysroot version.
+#[cargo_test(build_std_mock)]
+fn non_proc_macro_crate_uses_non_sysroot_proc_macro() {
+    let setup = setup();
+
+    let p = project()
+        .file(
+            "src/lib.rs",
+            r#"
+                extern crate proc_macro;
+                fn f() {
+                    let _ts = proc_macro::TokenStream::new();
+                }
+            "#,
+        )
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "pm"
+                version = "0.1.0"
+            "#,
+        )
+        .build();
+
+    p.cargo("build -v")
+        .build_std(&setup)
+        .target_host()
+        .run_expect_error();
```

Comment:
This should use `with_status()`, and also have `with_stderr()` to verify *which* error is happening.

Also, similar to the other questions, I'm not clear why this should fail. What's wrong with having a library access the proc_macro crate?